app/src/js/components/Table/Table.js

```diff
@@ -8,224 +8,209 @@ import omitBy from 'lodash/omitBy';
 import ErrorReport from '../Errors/report';
 import Loading from '../LoadingIndicator/loading-indicator';
 import Pagination from '../Pagination/pagination';
-import SortableTable from '../SortableTable/SortableTable';
 // Lodash
 import ListActions from '../ListActions/ListActions';
 import TableHeader from '../TableHeader/table-header';
 
+const SortableTable = lazy(() => import('../SortableTable/SortableTable'));
+
 function buildSortKey(sortProps) {
   return sortProps.filter((item) => item.id).map((item) => (item.desc === true ? `-${item.id}` : `+${item.id}`));
 }
 
-class List extends React.Component {
-  constructor(props) {
-    super(props);
-    this.queryNewPage = this.queryNewPage.bind(this);
-    this.queryNewSort = this.queryNewSort.bind(this);
-    this.updateSelection = this.updateSelection.bind(this);
-    this.onBulkActionSuccess = this.onBulkActionSuccess.bind(this);
-    this.onBulkActionError = this.onBulkActionError.bind(this);
-    this.getQueryConfig = this.getQueryConfig.bind(this);
-
-    const initialPage = 1;
-    const initialSortId = props.sortId;
-    const initialOrder = 'desc';
-    const sortProps = initialSortId ? [{ id: initialSortId, desc: true }] : [];
-
-    this.state = {
-      page: initialPage,
-      sortId: initialSortId,
-      order: initialOrder,
-      sortProps,
-      selected: [],
-      clearSelected: false,
-      infix: null,
-      queryConfig: {
-        page: initialPage,
-        sort_key: buildSortKey(sortProps),
-        ...(props.query || {}),
-      },
-      params: {},
-      completedBulkActions: 0,
-      bulkActionError: null,
-    };
-  }
-
-  componentDidUpdate(prevProps) {
-    const { list, query, queryParams } = this.props;
-
-    if (!isEqual(query, prevProps.query)) {
-      // eslint-disable-next-line react/no-did-update-set-state
-      this.setState({ queryConfig: this.getQueryConfig({}, query) });
-    }
-
+const List = ({
+  action,
+  bulkActions,
+  children,
+  data,
+  dispatch,
+  filterAction,
+  filterClear,
+  initialSortId,
```

Comment:
I'm not seeing sortId as an prop for List anymore, but I do see it though out the code. That seems wrong.

Reply:
Another oversight, updated the props to be initialSortId instead when rendering <List>